packages/core/core/src/ReporterRunner.js

```diff
@@ -113,6 +113,7 @@ export default class ReporterRunner {
           });
         } catch (reportError) {
           INTERNAL_ORIGINAL_CONSOLE.error(reportError);
+          process.exitCode = 1;
```

Comment:
I wonder if it is unexpected to do this for people using the Parcel programmatic API rather than our CLI? Should we store our own property here (rather than setting the global exit code), and then throw/emit an error from the main Parcel object that the CLI could pick up. We do already set an exit code in the CLI when an error is thrown.

Reply:
What if we just removed this `catch` and let the error propagate naturally?

Reply:
I was trying to avoid propagating the errors because not all of the calls are awaited on, and doing so could regress performance.

As for storing something that the CLI can pickup, I can try doing that but it will only get emitted at the end of the build unless the aforementioned un-awaited cases are changed.